libs/dao/src/main/java/com/akto/dto/ApiCollectionUsers.java

```diff
@@ -56,6 +68,108 @@ public enum CollectionType {
             });
         }});
 
+    public static Set<String> getUrlsFromConditions(List<CollectionCondition> conditions){
+        Set<ApiInfoKey> apis = new HashSet<>();
+        conditions.forEach((condition) -> {
+            if(CollectionCondition.Operator.AND.equals(condition.getOperator())){
+                apis.retainAll(condition.returnApis());
+            } else {
+                apis.addAll(condition.returnApis());
+            }
+        });
+
+        Set<String> urls = new HashSet<>();
+        apis.forEach((api) -> {
+            urls.add(api.getUrl() + " " + api.getMethod());
+        });
+        return urls;
+    }
+
+    public static void updateApiCollection(List<CollectionCondition> conditions, int id) {
+        Set<String> urls = getUrlsFromConditions(conditions);
+
+        ApiCollectionsDao.instance.updateOne(
+                Filters.eq(Constants.ID, id),
+                Updates.combine(
+                    Updates.set(ApiCollection.CONDITIONS_STRING, conditions),
+                    Updates.set(ApiCollection.URLS_STRING, urls)));
```

Comment:
This should be `addToSet` ? `Updates.set` will override existing variable in db.

Reply:
This function is used to add as well as remove APIs from a custom collection, thus using `Updates.set`